### PR TITLE
Initial round of passing lint tests in sg-core

### PR DESCRIPTION
**Pull request by leifmadsen**
Basing this on top of the current github-actions branch. This should be rebased to master once / if that merges (or changes).

### CODE REVIEW

**Comment by leifmadsen** [comment]
[Migrate from Travis-CI to GitHub Actions](https://trello.com/c/QVFdw42L/1210-migrate-from-travis-ci-to-github-actions)

File: pkg/collectd/collectd.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	jsoniter "github.com/json-iterator/go"
 )
 
-type CollectdMeta struct {
+type collectdMeta struct {
```
Comment (pleimer) [inline pkg/collectd/collectd.go:10]: 👍

The performance of this should probably be analysed more in depth. Golang object allocation can be expensive time-wise. This is not for this PR but I will some testing for the SG core refactor

**Review by pleimer — APPROVED** [review]
LGTM

File: pkg/collectd/collectd.go (hunk 1 above)
Comment (leifmadsen) [reply pkg/collectd/collectd.go:10]: I wonder if there is some profile tooling we could use to help us analyse these types of code paths?

...there is... I did some digging, and here are a few things we can look in to as part of the refactoring:

* https://goreportcard.com/report/github.com/infrawatch/sg-core
* https://blog.golang.org/pprof
* https://github.com/golangci/golangci-lint

**Pull request merged**